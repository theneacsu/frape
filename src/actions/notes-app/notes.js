import uuid from 'uuid/v4'

const addNote = ({ title, body = '' }) => ({
  type: 'ADD_NOTE',
  note: {
    title,
    body,
    id: uuid()
  }
})

const removeNote = id => ({
  type: 'REMOVE_NOTE',
  id
})

const editNote = (id, updates) => ({
  type: 'EDIT_NOTE',
  id,
  updates
})

export { addNote, removeNote, editNote }
