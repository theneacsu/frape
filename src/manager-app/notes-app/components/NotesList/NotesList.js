import React from 'react'
import { connect } from 'react-redux'
import NotesListItem from '../NotesListItem/NotesListItem'

const NotesList = props => (
  <div>
    {props.notes.length === 0 ? (
      <p>No notes. Get started by creating one.</p>
    ) : (
      props.notes.map(note => <NotesListItem key={note.id} {...note} />)
    )}
  </div>
)

const mapStateToProps = state => ({
  notes: state.notes
})

export default connect(mapStateToProps)(NotesList)
