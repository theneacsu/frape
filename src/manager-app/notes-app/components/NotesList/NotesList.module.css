.div {
  display: flex;
  flex-wrap: wrap;

  justify-content: center;
}

@media (min-width: 450px) {
  .div {
    justify-content: space-evenly;
  }
}
