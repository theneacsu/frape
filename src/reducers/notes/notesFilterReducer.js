const defaultState = {
  searchTerm: ''
}

const notesFilterReducer = (state = defaultState, action) => {
  switch (action.type) {
    case 'SET_SEARCH_TEXT':
      return {
        searchTerm: action.searchTerm
      }
    default:
      return state
  }
}

export default notesFilterReducer
