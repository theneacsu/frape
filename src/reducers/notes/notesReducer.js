const notesReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_NOTE':
      return [...state, action.note]
    case 'REMOVE_NOTE':
      return state.filter(note => note.id !== action.id)
    case 'EDIT_NOTE':
      return state.map(note => {
        if (note.id === action.id) {
          return {
            ...note,
            ...action.updates
          }
        }
        return note
      })
    default:
      return state
  }
}

export default notesReducer
