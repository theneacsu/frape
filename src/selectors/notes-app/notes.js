const getVisibleNotes = (notes, filters) => {
  if (filters.searchTerm) {
    return notes.filter(note => {
      const isTitleMatch = note.title
        .toLowerCase()
        .includes(filters.searchTerm.toLowerCase())
      const isBodyMatch = note.body
        .toLowerCase()
        .includes(filters.searchTerm.toLowerCase())

      return isTitleMatch || isBodyMatch
    })
  }
  return notes
}

export { getVisibleNotes }
