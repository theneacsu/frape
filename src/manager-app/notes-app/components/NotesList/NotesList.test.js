import React from 'react'
import { shallow } from 'enzyme'
import { NotesList } from './NotesList'
import notes from '../../../../coverage/fixtures/notes-app/notes'

test('Should render the NotesList component correctly with no notes provided', () => {
  const wrapper = shallow(<NotesList notes={[]} />)
  expect(wrapper).toMatchSnapshot()
})

test('Should render the NotesList component correctly with notes provided', () => {
  const wrapper = shallow(<NotesList notes={notes} />)
  expect(wrapper).toMatchSnapshot()
})
