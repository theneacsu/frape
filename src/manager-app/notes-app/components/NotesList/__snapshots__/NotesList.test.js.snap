// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the NotesList component correctly with no notes provided 1`] = `
<div
  className="div"
>
  <p>
    No notes. Get started by creating one.
  </p>
</div>
`;

exports[`Should render the NotesList component correctly with notes provided 1`] = `
<div
  className="div"
>
  <NotesListItem
    body="Try to cook your own food"
    id="1"
    key="1"
    title="Pizza is the devil"
  />
  <NotesListItem
    body="Tea is always a good idea"
    id="2"
    key="2"
    title="Drink more tea"
  />
  <NotesListItem
    body="Learn a new thing every day"
    id="3"
    key="3"
    title="Learn stuff"
  />
</div>
`;
